src/capturer/engine/win/mod.rs

```diff
@@ -143,40 +156,54 @@ pub fn create_capturer(options: &Options, tx: mpsc::Sender<Frame>) -> WinStream
         false => DrawBorderSettings::WithoutBorder,
     };
 
-    let settings = Settings::new(
-        WCMonitor::primary().unwrap(),
-        show_cursor,
-        show_highlight,
-        color_format,
-        FlagStruct {
-            tx,
-            crop: Some(get_crop_area(options)),
-        },
-    );
-
-    return WinStream {
+    let settings = match target {
+        Target::Display(display) => Settings::Display(WCSettings::new(
+            WCMonitor::from_raw_hmonitor(display.raw_handle.0),
+            show_cursor,
+            show_highlight,
+            color_format,
+            FlagStruct {
+                tx,
+                crop: Some(get_crop_area(options)),
+            },
+        )),
+        Target::Window(window) => Settings::Window(WCSettings::new(
+            WCWindow::from_raw_hwnd(window.raw_handle.0),
+            show_cursor,
+            show_highlight,
+            color_format,
+            FlagStruct {
+                tx,
+                crop: Some(get_crop_area(options)),
+            },
+        )),
+    };
+
+    WCStream {
         settings,
         capture_control: None,
-    };
+    }
 }
 
 pub fn get_output_frame_size(options: &Options) -> [u32; 2] {
-    // TODO: this should be based on display from options.target, not main one
-    let display = targets::get_main_display();
-    let display_id = display.id;
-    let scale_factor = targets::get_scale_factor(display_id);
+    let target = options
+        .target
```

Comment:
What happens if the user does not pass a target?
I think we should use the main display if the user does not pass a target instead of forcing the user to pass a target. Thoughts?

Reply:
> I think we should use the main display

That is exactly the current behaviour. The last `unwrap_or_else` on L192 accounts for this.